pkg/router/controller/route_secret_manager.go

```diff
@@ -28,18 +38,33 @@ type RouteSecretManager struct {
 
 	secretManager secretmanager.SecretManager
 	secretsGetter corev1client.SecretsGetter
+	routelister   routelisters.RouteLister
 	sarClient     authorizationclient.SubjectAccessReviewInterface
+	// deletedSecrets tracks routes for which the associated secret was deleted after intial creation of the secret monitor.
+	// This helps to differentiate between a new secret creation and a recreation of a previously deleted secret.
+	// Populated inside DeleteFunc, and consumed or cleaned inside AddFunc and unregister().
+	// It is thread safe and "namespace/routeName" is used as it's key.
```

Comment:
```suggestion
	// It is thread safe and "namespace/routeName" is used as its key.
```

Reply:
Updated in https://github.com/openshift/router/commit/62daef459fde17c27007cd8d0ac6bc24209b2120